pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -72,21 +72,8 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, pr *v1beta1.PipelineRun)
 
 	// Get TaskRun names depending on whether embeddedstatus feature is set or not
 	var trs []string
-	if len(pr.Status.ChildReferences) == 0 || len(pr.Status.TaskRuns) > 0 || len(pr.Status.Runs) > 0 { //nolint:all //incompatible with pipelines v0.45
-		for trName, ptrs := range pr.Status.TaskRuns { //nolint:all //incompatible with pipelines v0.45
-			// TaskRuns within a PipelineRun may not have been finalized yet if the PipelineRun timeout
-			// has exceeded. Wait to process the PipelineRun on the next update, see
-			// https://github.com/tektoncd/pipeline/issues/4916
-			if ptrs.Status == nil || ptrs.Status.CompletionTime == nil {
-				logging.FromContext(ctx).Infof("taskrun %s within pipelinerun is not yet finalized: embedded status is not complete", trName)
-				return nil
-			}
-			trs = append(trs, trName)
-		}
```

Comment:
This looks like this was added intentionally - is this safe to remove? 🤔 cc @lcarva

Reply:
`.Status.TaskRuns` and `.Status.Runs` don't exist in v1. So I think replacing this if-else-block with the for-loop below is correct. `.Status.ChildReferences` is the only way to get the status of the TaskRun from the PipelineRun. :+1:

Reply:
+1 to what @lcarva mentioned - these fields `.Status.TaskRuns` and `.Status.Runs` no longer exist in the v1 objects so we can only use `.Status.ChildReferences` checks now